flyteplugins/go/tasks/pluginmachinery/internal/webapi/core.go

```diff
@@ -99,6 +99,13 @@ func (c CorePlugin) Handle(ctx context.Context, tCtx core.TaskExecutionContext)
 }
 
 func (c CorePlugin) Abort(ctx context.Context, tCtx core.TaskExecutionContext) error {
+	cacheItemID := tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName()
+	err := c.cache.DeleteDelayed(cacheItemID)
+	if err != nil {
+		logger.Errorf(ctx, "Failed to delete resource [%v] from cache. Error: %v", cacheItemID, err)
+		return err
+	}
+
```

Comment:
Should we do this in Finalize? so that it happens all the time

Reply:
done